### PR TITLE
FROM-less selects support

### PR DESCRIPTION
**Pull request by ivanthescientist**
Possible solution for #18580.

Enables `FROM`-less `SELECT`s at the parser-validator level, engine itself already supports it.

### CODE REVIEW

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SelectWithoutFromStatementTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+
+import static org.junit.Assert.assertEquals;
+
+public class SelectWithoutFromStatementTest extends SqlTestSupport {
```
Comment (gierlachg) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SelectWithoutFromStatementTest.java:35]: Maybe we could assert result `SqlRowMetadata` as well.

**Review by gierlachg — APPROVED** [review]

**Review by viliam-durina — APPROVED** [review]

**Review by hakanmemisoglu — APPROVED** [review]

**Comment by ivanthescientist** [comment]
run-lab-run

**Comment by ivanthescientist** [comment]
run-lab-run

**Comment by ivanthescientist** [comment]
run-lab-run

**Comment by ivanthescientist** [comment]
run-lab-run

**Pull request merged**